### PR TITLE
[docs] Refactor docs - Test Organization

### PR DESCRIPTION
**Pull request by VasilyStrelyaev**

### CODE REVIEW

File: docs/articles/documentation/reference/test-api/testcontroller/setpageloadtimeout.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+t.setPageLoadTimeout( duration )
+```
+
+Parameter  | Type      | Description
+---------- | --------- | -----------
+`duration` | Number    | Page load timeout (in milliseconds). `0` to skip waiting for the `window.load` event.
+
+After the timeout passes or the `window.load` event is raised (whichever happens first), TestCafe starts the test.
+
+You can also set the page load timeout when you launch tests from the [command line](../../command-line-interface.md#--page-load-timeout-ms) or [API](../../programming-interface/runner.md#run).
```
Comment (anthophobiac) [inline docs/articles/documentation/reference/test-api/testcontroller/setpageloadtimeout.md]: >or [API]

or [the API]
or the [API]

File: docs/articles/documentation/reference/test-api/testcontroller/README.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+            .click(this.signInButton);
+    }
+}
+
+export default new Page();
+```
+
+In this example, the page model's `login` method uses the test controller to perform authentication actions.
+
+TestCafe can implicitly resolve the test context when you import the test controller so you do not have to pass the test controller object explicitly.
```
Comment (anthophobiac) [inline docs/articles/documentation/reference/test-api/testcontroller/README.md]: >TestCafe can implicitly resolve the test context when you import the test controller so you do not have to pass the test controller object explicitly.

TestCafe can implicitly resolve the test context when you import the test controller, so you do not have to pass the test controller object explicitly.

A comma is missing before `so`.

File: docs/articles/documentation/reference/test-api/test/skip.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
+---
+layout: docs
+title: Test.skip Method
+permalink: /documentation/reference/test-api/test/skip.html
+---
+# Test.skip Method
+
+Specifies that TestCafe should not run this test.
```
Comment (anthophobiac) [inline docs/articles/documentation/reference/test-api/test/skip.md]: >Specifies that TestCafe should not run this test.

Specifies that TestCafe should not run the test.

File: docs/articles/documentation/guides/basic-guides/test-organization.md [hunk 4]
```diff
@@ -0,0 +1,271 @@
...
+test('Test2', async t => {
+    /* Test 2 Code */
+});
+```
+
+The test code function accepts the [test controller](../../reference/test-api/testcontroller/README.md) object as a parameter. The test controller provides access to TestCafe test API.
+
+You can arrange test code in any manner and reference any modules or libraries.
+
+TestCafe tests are executed on the server side. You can use [test actions](interact-with-the-page.md) to manipulate the tested webpage. To determine page elements' state or obtain any other data from the client side, use the [selectors](select-page-elements.md) and [client functions](obtain-client-side-info.md).
```
[the remark this answers is not in the document]
Comment (VasilyStrelyaev) [reply docs/articles/documentation/guides/basic-guides/test-organization.md]: Mmm... are you sure (Grammarly is, but are you)? According to how you would normally use hyphens, it is *executed on the client side* but *a client-side function*.

[the remark this answers is not in the document]
Comment (VasilyStrelyaev) [reply docs/articles/documentation/guides/basic-guides/test-organization.md]: Same as above.

File: docs/articles/documentation/guides/basic-guides/test-organization.md [hunk 5]
```diff
@@ -0,0 +1,271 @@
...
+fixture `Fixture 2`;
+
+test('Fixture 2 - Test 1', () => {});
+test.only('Fixture 2 - Test 2', () => {});
+test('Fixture 2 - Test 3', () => {});
+
+// Only tests in 'Fixture 1' and the 'Fixture 2 - Test 2' test are run
+```
+
+If several tests or fixtures are marked with `only`, all the marked tests and fixtures are run.
```
[the remark this answers is not in the document]
Comment (VasilyStrelyaev) [reply docs/articles/documentation/guides/basic-guides/test-organization.md]: I think it's okay. Marking several tests with `only` is an edge case, and we shouldn't overburden the first sentence with such facts.

File: docs/articles/documentation/reference/test-api/fixture/clientscripts.md [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+fixture
+    .clientScripts({
+        page: /\/user\/profile\//,
+        content: 'Geolocation.prototype.getCurrentPosition = () => new Positon(0, 0);'
+    });
+```
+
+> Relative paths are resolved against the test file location.
+
+You can use the [page](../../../guides/advanced-guides/inject-client-scripts.md#provide-scripts-for-specific-pages) option to specify pages into which scripts should be injected. Otherwise, TestCafe injects scripts into all pages visited during the test.
```
[the remark this answers is not in the document]
Comment (VasilyStrelyaev) [reply docs/articles/documentation/reference/test-api/fixture/clientscripts.md]: I think 'visited' is more clear, because I could think that 'pages where the tests run' means start pages.

As for 'missing', okay, I'll use it, and let's see what happens.

File: docs/articles/documentation/reference/test-api/fixture/skip.md [hunk 7]
```diff
@@ -0,0 +1,27 @@
+---
+layout: docs
+title: Fixture.skip Method
+permalink: /documentation/reference/test-api/fixture/skip.html
+---
+# Fixture.skip Method
+
+Specifies that TestCafe should not run this fixture.
```
[the remark this answers is not in the document]
Comment (VasilyStrelyaev) [reply docs/articles/documentation/reference/test-api/fixture/skip.md]: Okay, just not 'specified' I think, because we are talking about the current fixture. So, 'the fixture' I guess.

File: docs/articles/documentation/guides/basic-guides/test-organization.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (anthophobiac) [reply docs/articles/documentation/guides/basic-guides/test-organization.md]: @VasilyStrelyaev  I don't simply rely on Grammarly, y'know! I just saw the hyphenated option generally used more often in documentation and support tickets.

File: docs/articles/documentation/reference/test-api/fixture/page.md [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+```js
+fixture `MyFixture`
+    .page `http://devexpress.github.io/testcafe/example`;
+
+test('Test1', async t => {
+    // Starts at http://devexpress.github.io/testcafe/example
+});
+```
+
+You can also specify a start page for individual tests with the [test.page](../test/page.md) function that overrides `fixture.page`.
```
Comment (arminal) [inline docs/articles/documentation/reference/test-api/fixture/page.md]:
```suggestion
You can also specify a start page for individual tests with the [test.page](../test/page.md) function that overrides the `fixture.page`.
```

File: docs/articles/documentation/reference/test-api/test/disablepagecaching.md [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+
+**Example**
+
+```js
+test
+    .disablePageCaching
+    ('My test', async t => { /* ... */ });
+```
+
+You can also disable page caching during all test in a fixture with the [fixture.disablePageCaching](../fixture/disablepagecaching.md) method.
```
Comment (arminal) [inline docs/articles/documentation/reference/test-api/test/disablepagecaching.md]:
```suggestion
You can also disable page caching during all tests in a fixture with the [fixture.disablePageCaching](../fixture/disablepagecaching.md) method.
```

File: docs/articles/documentation/reference/test-api/test/only.md [hunk 10]
```diff
@@ -0,0 +1,30 @@
+---
+layout: docs
+title: Test.only Method
+permalink: /documentation/reference/test-api/test/only.html
+---
+# Test.only Method
+
+Specifies that TestCafe should run only this test and skip all other tests.
```
Comment (arminal) [inline docs/articles/documentation/reference/test-api/test/only.md]:
```suggestion
Specifies that TestCafe should skip all other tests and run only this test.
```

File: docs/articles/documentation/reference/test-api/testcontroller/README.md [hunk 11]
```diff
@@ -0,0 +1,76 @@
...
+        .click('#populate')
+        .click('#submit-button');
+
+    const location = await t.eval(() => window.location);
+
+    await t.expect(location.pathname).eql('/testcafe/example/thank-you.html');
+});
+```
+
+The test runner also uses the test controller to access the internal context required for the test API to operate. When you call [selectors](select-page-elements.md) and [client functions](obtain-client-side-info.md) from Node.js callbacks, pass the test controller explicitly because the API cannot retrieve it from the context.
```
Comment (arminal) [inline docs/articles/documentation/reference/test-api/testcontroller/README.md]:
```suggestion
The test runner also uses the test controller to access the internal context required for the test API to operate. When you call [selectors](select-page-elements.md) and [client functions](obtain-client-side-info.md) from Node.js callbacks, pass the test controller explicitly, because the API cannot retrieve it from the context.
```

File: docs/articles/documentation/reference/test-api/testcontroller/fixturectx.md [hunk 12]
```diff
@@ -0,0 +1,44 @@
...
+fixture `Fixture1`
+    .before(async ctx  => {
+        ctx.someProp = 123;
+    })
+    .after(async ctx  => {
+        console.log(ctx.someProp); // > 123
+    });
+```
+
+Test code can read the fixture context from `t.fixtureCtx`, assign to its properties or add new ones, but it cannot overwrite the entire `t.fixtureCtx` object.
```
Comment (arminal) [inline docs/articles/documentation/reference/test-api/testcontroller/fixturectx.md]: assign what to its properties?

File: docs/articles/documentation/guides/basic-guides/test-organization.md [hunk 13]
```diff
@@ -0,0 +1,270 @@
...
+});
+```
+
+The test code function accepts the [test controller](../../reference/test-api/testcontroller/README.md) object as a parameter. The test controller provides access to the TestCafe test API.
+
+You can arrange test code in any manner and reference any modules or libraries.
+
+TestCafe tests are executed on the server side. You can use [test actions](interact-with-the-page.md) to manipulate the tested webpage. To determine page elements' state or obtain any other data from the client side, use the [selectors](select-page-elements.md) and [client functions](obtain-client-side-info.md).
+
+To check if the page state matches the expected one, use [assertions](assert.md).
```
[the remark this answers is not in the document]
Comment (VasilyStrelyaev) [reply docs/articles/documentation/guides/basic-guides/test-organization.md]: Would the following be okay:
*To check if the page is in the expected state, use [assertions](assert.md).*
or
*To check if page elements have the expected parameters, use [assertions](assert.md).*

File: docs/articles/documentation/guides/basic-guides/test-organization.md [hunk 14]
```diff
@@ -0,0 +1,270 @@
...
+test
+    .page `http://devexpress.github.io/testcafe/blog/`
+    ('My test', async t => {
+        // Starts at http://devexpress.github.io/testcafe/blog/
+    });
+```
+
+## Specify Test Metadata
+
+TestCafe allows you to specify additional information for tests in the form of *key-value metadata* and use it to filter tests or display in the reports.
```
[the remark this answers is not in the document]
Comment (VasilyStrelyaev) [reply docs/articles/documentation/guides/basic-guides/test-organization.md]: *TestCafe allows you to specify additional information for tests in the form of *key-value metadata*. You can display this information in the reports and use it to filter tests.*

File: docs/articles/documentation/reference/test-api/testcontroller/fixturectx.md (hunk 12 above)
Comment (VasilyStrelyaev) [reply docs/articles/documentation/reference/test-api/testcontroller/fixturectx.md]: *Test code can read the fixture context from `t.fixtureCtx`, assign values to its properties or add new properties, but it cannot overwrite the entire `t.fixtureCtx` object.*

File: docs/articles/documentation/guides/basic-guides/test-organization.md (hunk 13 above)
[the remark this answers is not in the document]
Comment (arminal) [reply docs/articles/documentation/guides/basic-guides/test-organization.md]: I like the second version.

File: docs/articles/documentation/guides/basic-guides/test-organization.md (hunk 14 above)
[the remark this answers is not in the document]
Comment (arminal) [reply docs/articles/documentation/guides/basic-guides/test-organization.md]: Yes, that works.

File: docs/articles/documentation/reference/test-api/testcontroller/fixturectx.md (hunk 12 above)
Comment (arminal) [reply docs/articles/documentation/reference/test-api/testcontroller/fixturectx.md]: ok good
